lib/apiservers/engine/proxy/archive.go

```diff
@@ -32,7 +32,7 @@ import (
 
 type VicArchiveProxy interface {
 	ArchiveExportReader(ctx context.Context, store, ancestorStore, deviceID, ancestor string, data bool, filterSpec vicarchive.FilterSpec) (io.ReadCloser, error)
-	ArchiveImportWriter(ctx context.Context, store, deviceID string, filterSpec vicarchive.FilterSpec) (io.WriteCloser, error)
+	ArchiveImportWriter(ctx context.Context, store, deviceID string, filterSpec vicarchive.FilterSpec, errchan chan error) (io.WriteCloser, error)
```

Comment:
The reason I passed this in is that I didn't want to wait on a variable number of channels to unblock. I could use a wait group but I will still have to allocate buffer and give index to each stream to store their returned error. So I am using one buffered error channel with enough error entries for maximum number of streams.